tests/compile-fail/function_calls/check_callback_abi.rs

```diff
@@ -0,0 +1,15 @@
+#![feature(core_intrinsics)]
+
+extern "C" fn try_fn(_: *mut u8) {
+    unreachable!();
+}
+
+fn main() {
+    unsafe {
+        std::intrinsics::r#try( //~ ERROR calling a function with ABI C using caller ABI Rust
+            std::mem::transmute::<extern "C" fn(*mut u8), _>(try_fn),
+            std::ptr::null_mut(),
+            |_, _| unreachable!(),
+        );
+    }
+}
```

Comment:
(Adding this because I didn't find any test other than `tests/compile-fail/concurrency/unwind_top_of_stack.rs` exercising the ABI check in `call_function()`, and that test is changed to test "unwinding past the topmost frame of the stack" instead of the ABI mismatch now.)